tst/EnergyPlus/unit/HVACDXSystem.unit.cc

```diff
@@ -571,4 +572,151 @@ TEST_F(EnergyPlusFixture, VariableSpeedCoils_RHControl)
+        "    1Cap,                    !- Speed 5 Total Cooling Capacity Function of Temperature Curve Name",
+        "    CAPFF,                   !- Speed 5 Total Cooling Capacity Function of Air Flow Fraction Curve Name",
+        "    1Pow,                    !- Speed 5 Energy Input Ratio Function of Temperature Curve Name",
+        "    EIRFF;                   !- Speed 5 Energy Input Ratio Function of Air Flow Fraction Curve Name",
+
+        "Curve:Quadratic, PLFCurve, 0.85, 0.83, 0.0, 0.0, 0.3, 0.85, 1.0, Dimensionless, Dimensionless; ",
+        "Curve:Cubic, CAPFF, 1, 0, 0, 0, 0, 1, , , Dimensionless, Dimensionless; ",
+        "Curve:Cubic, EIRFF, 1, 0, 0, 0, 0, 1, , , Dimensionless, Dimensionless; ",
+        "Curve:Biquadratic, 1Cap, 0.483, 0.0305, 0.0000458, 0.00511, -1.50E-04, -1.28E-04, 8.89, 21.67, 12.78, 51.67, , , Temperature, Temperature, "
+        "Dimensionless; ",
+        "Curve:Biquadratic, 1Pow, 1.33, -0.034, 0.00094, -0.0086, 0.00077, -0.000972, 8.89, 21.7, 12.8, 51.7, , , Temperature, Temperature, "
+        "Dimensionless; ",
+
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    DataGlobals::NumOfTimeStepInHour = 1;
+    DataGlobals::MinutesPerTimeStep = 60;
+    OutputReportPredefined::SetPredefinedTables();
+    ScheduleManager::ProcessScheduleInput(state.files);
+    ScheduleManager::Schedule(1).CurrentValue = 1.0; // Enable schedule without calling schedule manager
+
+    int DXSystemNum = 1;
+    bool FirstHVACIteration = true;
+    bool HXUnitOn = false;
+    int InletNode = 1;
+    int ControlNode = 2; // same as outlet node number
+
+    HVACDXSystem::GetDXCoolingSystemInput(state);
+
+    // set up outdoor environment
+    DataEnvironment::OutDryBulbTemp = 35.0;
+    DataEnvironment::OutHumRat = 0.0196;
+    DataEnvironment::OutBaroPress = 101325.0;
+    DataEnvironment::OutWetBulbTemp = 27.0932;
+
+    // set up inputs to test coil control
+    HVACDXSystem::DXCoolingSystem(DXSystemNum).DesiredOutletTemp = 21.0;
+    DataEnvironment::StdRhoAir = 1.2;
+    DataLoopNode::Node(InletNode).MassFlowRate = 5.66336932 * DataEnvironment::StdRhoAir;
+    DataLoopNode::Node(InletNode).Temp = 24.0;
+    //DataLoopNode::Node(InletNode).HumRat = 0.012143698;
+    DataLoopNode::Node(InletNode).HumRat = 0.014;
+    DataLoopNode::Node(InletNode).Enthalpy =
+        Psychrometrics::PsyHFnTdbW(DataLoopNode::Node(InletNode).Temp, DataLoopNode::Node(InletNode).HumRat); // 55029.3778; // conditions at 65 % RH
+    DataLoopNode::Node(ControlNode).TempSetPoint = HVACDXSystem::DXCoolingSystem(DXSystemNum).DesiredOutletTemp;
+
+    // test sensible control
+    HVACDXSystem::ControlDXSystem(state, DXSystemNum, FirstHVACIteration, HXUnitOn);
+    Real64 SHR = VariableSpeedCoils::VarSpeedCoil(1).QSensible / VariableSpeedCoils::VarSpeedCoil(1).QLoadTotal;
+    EXPECT_NEAR(SHR, 0.811509, 0.0001);
+    DataLoopNode::Node(InletNode).MassFlowRate = 1.61576817;
```

Comment:
The previous coil inputs resulted in an SHR = 0.811509 without latent degradation. Adding latent degradation would not change the SHR much, if at all, at this high PLR. Then the mass flow rate is changed and latent degradation inputs added. The resulting SHR = 1, but what is that compared to? It's at a different mass flow rate. What was the SHR before the latent degradation inputs were added using the new mass flow rate? Or just adjust the coil inputs or set point to give a lower SHR before adding the latent degradation inputs. The point here is that this unit test doesn't really test the latent degradation model.

Reply:
@rraustad I agree with your opinion that the unit test is not very meaningful. The unit test is based on the original calculation of QLatRated
                    QLatRated = QLoadTotal - QSensible;

If QLatRated is calculated based on my revision from its definition: 

                    QLatRated = VarSpeedCoil(DXCoilNum).MSRatedTotCap(SpeedCal) * (1.0 - VarSpeedCoil(DXCoilNum).MSRatedSHR(SpeedCal));

the meaningful unit test is obtained from my previous check-in:

    //DataLoopNode::Node(InletNode).MassFlowRate = 1.61576817;
    VariableSpeedCoils::VarSpeedCoil(1).Twet_Rated = 1000.0;
    VariableSpeedCoils::VarSpeedCoil(1).Gamma_Rated = 1.5;
    HVACDXSystem::ControlDXSystem(state, DXSystemNum, FirstHVACIteration, HXUnitOn);
    SHR = VariableSpeedCoils::VarSpeedCoil(1).QSensible / VariableSpeedCoils::VarSpeedCoil(1).QLoadTotal;
    //EXPECT_NEAR(SHR, 1.0, 0.0001);
    EXPECT_NEAR(SHR, 0.878975, 0.0001);

When I ran a test file using two different QLatRated calculations, the differences are minor, as shown in the previously attached file.

As mentioned before, if conditions are OK, the original calculation should get the same results as one from its definition. However, the same results would be very rare in simulations. The calculation results of my revision are based on inputs. The conditions needed to get same results from both calculations are:

1. Indoor wet bulb temperature and outdoor dry bulb temperature are at rated conditions
2. All multipliers at rated conditions are equal to 1.0

Otherwise, different results are obtained.

In order to get meaningful unit test, my revision should be used. That is a reason I made the change,